### PR TITLE
External authentication support

### PR DESCRIPTION
**Pull request by aChanEP**
## Type

* ### Feature

Adds three new resources to the SDK in order to support oidc

1. Adds integration to the Profiles resource
2. Adds integration to the Realms resource
3. Adds integration to the authentication settings resource

Other changes:
Changed the parameters that we can give to the config so that we can specify the protocol for the endpoint it connects to as well as a storeID.

* ### Fix
  Fixes a bug
* ### Docs
  Documentation only changes
* ### Style
  Changes that **do not** affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
* ### Refactor
  A code change that neither fixes a bug nor adds a feature
* ### Performance
  A code change that improves performance
* ### Test
  Adding missing or correcting existing tests
* ### Chore
  Changes to the build process or auxiliary tools and libraries such as documentation generation  

## Description

*A brief description of the goals of the pull request.*

## Dependencies

*Other PRs or builds that this PR depends on.*

## Issues

*A list of issues closed by this PR.*

* Fixes #

## Notes

*Any additional notes.*

### CODE REVIEW

File: src/endpoints/customers.js [hunk 1]
```diff
@@ -7,12 +7,32 @@ class CustomersEndpoint extends CRUDExtend {
     this.endpoint = 'customers'
   }
 
-  Token(email, password) {
-    return this.request.send(`${this.endpoint}/tokens`, 'POST', {
-      email,
-      password,
-      type: 'token'
-    })
+  Token(email, password, code, redirectUri, headers) {
```
Comment (aChanEP) [inline src/endpoints/customers.js:10]: refactored

File: src/factories/request.js [hunk 2]
```diff
@@ -42,7 +42,13 @@ class RequestFactory {
       body.client_secret = config.client_secret
     }
 
+    console.log('about to authenticate using the SDK!')
+
     const promise = new Promise((resolve, reject) => {
+      console.log(`${config.protocol}://${config.host}/${config.auth.uri}`)
```
Comment (aChanEP) [inline src/factories/request.js:48]: remove these console debug statements

File: src/types/address.d.ts [hunk 3]
```diff
@@ -60,3 +60,4 @@ export interface AddressesEndpoint {
 
   Delete(options: { customer: string, address: string, token?: string }): Promise<{}>;
 }
+
```
Comment (aChanEP) [inline src/types/address.d.ts:63]: remove this extra space

File: src/types/authentication-profile.d.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  meta: {
+    timestamps: {
+      created_at: string;
+      updated_at: string;
+    };
+  };
+  links: {}
+}
+
+/**
```
Comment (aChanEP) [inline src/types/authentication-profile.d.ts:29]: Will tend to this in a subsequent PR when docs are ready!

File: src/types/authentication-realm.d.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  id: string;
+  name: string;
+}
+
+export interface RealmCreateBody {
+  type: string;
+  name: string;
+}
+
+/**
```
Comment (aChanEP) [inline src/types/authentication-realm.d.ts:23]: will tend to this when docs are ready in another PR

File: src/types/authentication-settings.d.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+export interface AuthenticationSettingsBase {
+  type: string;
+  meta: object;
+  relationships: {
+      "authentication-realm": RelationshipToOne<'authentication-realm'>;
+    };
+}
+
+/**
```
Comment (aChanEP) [inline src/types/authentication-settings.d.ts:11]: will tend to this when docs are ready in another PR

File: test/unit/authentication-profile.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  it('Update a single authentication Profile', () => {
+    nock(apiUrl, {})
+      .put(/\/authentication-realms\/(.*)\/profiles\/oidc\/(.*)/)
+      .reply(201, {})
+
+    const body = {
+      name: 'Keycloak 2',
+      type: 'authentication-profile-oidc',
+      discovery_url:
+        'http://localhost:24074/auth/realms/Sample/.well-known/openid-configuration',
```
Comment (steve-r-west) [inline test/unit/authentication-profile.ts:68]: Should this be here?

File: src/types/authentication-profile.d.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  name: string;
+  type: string;
+  discovery_url: string;
+  client_id: string;
+  client_secret: string;
+}
+
+export interface Profile extends Resource<ProfileBase> {
+  links: {
+    "authentication-link": string,
```
Comment (steve-r-west) [inline src/types/authentication-profile.d.ts:27]:
```suggestion
    "authorization-endpoint": string,
```

**Pull request closed without merging**